plugins/woocommerce/src/Internal/Loggers/RemoteLogger.php

```diff
@@ -85,24 +90,36 @@ private function is_latest_woocommerce_version() {
 	 * @return string|null
 	 */
 	private function fetch_latest_woocommerce_version() {
-		$transient_key  = 'latest_woocommerce_version';
-		$cached_version = get_transient( $transient_key );
+		$cached_version = get_transient( self::WC_LATEST_VERSION_TRANSIENT );
 		if ( $cached_version ) {
 			return $cached_version;
 		}
 
+		$retry_count = get_transient( self::FETCH_LATEST_VERSION_RETRY );
+		if ( false === $retry_count ) {
```

Comment:
```suggestion
		if ( false === $retry_count || ! is_numeric ( $retry_count ) ) {
```

We could also do a numeric check in case the value is corrupted somehow